### PR TITLE
[12.0][MIG] sale_order_invoicing_finished_task: Migration to 12.0

### PR DESCRIPTION
**Pull request by Tardo**
Cc @Tecnativa

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Thanks, @Tardo ! 

Code and functional review :+1:

**Review by tbaden — APPROVED** [review]
thanks for the work 👍 
I did a code review: LGTM

File: sale_order_invoicing_finished_task/models/project.py [hunk 1]
```diff
@@ -59,8 +59,8 @@ def _check_sale_line_state(self, sale_line_id=False):
         if sale_line_id:
             sale_lines |= self.env['sale.order.line'].browse(sale_line_id)
         for sale_line in sale_lines:
-            if (sale_line.state in ('done', 'cancel') or
-                    sale_line.invoice_status == 'invoiced'):
+            if (sale_line.state in ('done', 'cancel')
+                    or sale_line.invoice_status == 'invoiced'):
```
Comment (pedrobaeza) [inline sale_order_invoicing_finished_task/models/project.py:63]: Please don't change this style for nothing. Leave it the one it has.

File: sale_order_invoicing_finished_task/models/sale_order.py [hunk 2]
```diff
@@ -33,23 +33,20 @@ class SaleOrderLine(models.Model):
                  'order_id.state', 'task_ids.invoiceable')
     def _get_to_invoice_qty(self):
         lines = self.filtered(
-            lambda x: (x.product_id.type == 'service' and
-                       x.product_id.invoicing_finished_task and
-                       x.product_id.service_tracking in [
-                           'task_global_project', 'task_new_project'] and
-                       not all(x.task_ids.mapped('invoiceable'))
+            lambda x: (x.product_id.type == 'service'
+                       and x.product_id.invoicing_finished_task
```
Comment (pedrobaeza) [inline sale_order_invoicing_finished_task/models/sale_order.py:37]: Please don't change this style for nothing. Leave it the one it has.

File: sale_order_invoicing_finished_task/models/sale_order.py [hunk 3]
```diff
@@ -33,23 +33,20 @@ class SaleOrderLine(models.Model):
...
+                       and not all(x.task_ids.mapped('invoiceable'))
                        )
         )
         if lines:
             lines.update({'qty_to_invoice': 0.0})
         super(SaleOrderLine, self - lines)._get_to_invoice_qty()
 
     @api.multi
-    def _analytic_compute_delivered_quantity_domain(self):
-        # Overwrite until resolve https://github.com/odoo/odoo/issues/24038
```
Comment (pedrobaeza) [inline sale_order_invoicing_finished_task/models/sale_order.py]: Why removing this comment if this still overwriting full domain? Can't you do this inheritable?

File: sale_order_invoicing_finished_task/readme/CONTRIBUTORS.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+* Denis Leemann <denis.leemann@camptocamp.com>
+* Sergio Teruel <sergio.teruel@tecnativa.com>
```
Comment (pedrobaeza) [inline sale_order_invoicing_finished_task/readme/CONTRIBUTORS.rst]: You can put the header with the company name and the URL, and remove emails.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_order_invoicing_finished_task/models/project.py (hunk 1 above)
Comment (Tardo) [reply sale_order_invoicing_finished_task/models/project.py:63]: This change was made to follow this rule:
https://lintlyci.github.io/Flake8Rules/rules/W504.html

**Comment by Tardo** [comment]
Changes made, except operator position... use the rule W504 or W503 (deprecated)?

File: sale_order_invoicing_finished_task/tests/test_sale_order_invoicing_finished_task.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+class TestInvoicefinishedTask(common.SavepointCase):
+
+    def setUp(self):
+        super(TestInvoicefinishedTask, self).setUp()
+        self.hour_uom = self.env.ref('uom.product_uom_hour')
+        self.env.user.company_id.project_time_mode_id = self.hour_uom.id
+        group_manager = self.env.ref('sales_team.group_sale_manager')
+        self.manager = self.env['res.users'].create({
+            'name': 'Andrew Manager',
+            'login': 'manager',
```
Comment (victor-ditay) [inline sale_order_invoicing_finished_task/tests/test_sale_order_invoicing_finished_task.py:18]:
```suggestion
            'login': 'test_manager',
```
Travis is failing due:

ERROR: duplicate key value violates unique constraint "res_users_login_key"
DETAIL:  Key (login)=(manager) already exists.

Comment (Tardo) [reply sale_order_invoicing_finished_task/tests/test_sale_order_invoicing_finished_task.py:18]: Changed, thanks!

**Comment by Tardo** [comment]
Ok, finally travis is happy again... the problem was the domain.

File: sale_order_invoicing_finished_task/views/project_view.xml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        </field>
+    </field>
+</record>
+
+<record id="edit_project_task_track" model="ir.ui.view">
+    <field name="name">project.task.form.track</field>
+    <field name="model">project.task</field>
+    <field name="inherit_id" ref="project.view_task_form2"/>
+    <field name="arch" type="xml">
+        <xpath expr="//field[@name='planned_hours']" position="after">
```
Comment (katyukha) [inline sale_order_invoicing_finished_task/views/project_view.xml]: Please, apply this PR #825 in this migration.

Comment (Tardo) [reply sale_order_invoicing_finished_task/views/project_view.xml]: Waiting for that PR to be approved to merge it here

Comment (pedrobaeza) [reply sale_order_invoicing_finished_task/views/project_view.xml]: You can apply it, @Tardo

Comment (Tardo) [reply sale_order_invoicing_finished_task/views/project_view.xml]: Done

Comment (katyukha) [reply sale_order_invoicing_finished_task/views/project_view.xml]: Thank you)

**Review by pedrobaeza — APPROVED** [review]

**Pull request closed without merging**